platform/ios/src/MGLMapView.h

```diff
@@ -286,6 +301,21 @@ MGL_EXPORT IB_DESIGNABLE
  */
 - (IBAction)showAttribution:(id)sender;
 
+/**
+ The preferred frame rate at which the map view is rendered.
+
+ The default value for this property is
+ `MGLMapViewPreferredFramesPerSecondDefault`, which will adaptively set the
+ preferred frame rate based on the capability of the user’s device to maintain
+ a smooth experience.
+
+ In addition to the provided `MGLMapViewPreferredFramesPerSecond` options, this
+ property can be set to arbitrary integer values.
+
+ @see `CADisplayLink.preferredFramesPerSecond`
+ */
+@property (nonatomic, assign) MGLMapViewPreferredFramesPerSecond preferredFramesPerSecond;
```

Comment:
What is the purpose for letting devs set 0...60? does it makes a difference than restricting to -1/30/60?

Reply:
Allowing developers to set arbitrary values means that we can allow:

- 120 FPS (via `0`) without explicitly documenting it.
- Extreme throttling, [as our navigation SDK currently does](https://github.com/mapbox/mapbox-navigation-ios/blob/5d746e13afc47c74eb78188283c9d3f316d1e7f6/MapboxNavigation/NavigationMapView.swift#L15-L21) (as low as 5 FPS, apparently).

The `-1` of `MGLMapViewPreferredFramesPerSecondDefault` is a placeholder so that we can name the key, but it has no direct meaning in terms of FPS (https://github.com/mapbox/mapbox-gl-native/pull/12501#discussion_r205917369).